client/root.js

```diff
@@ -7,60 +7,77 @@ import page from 'page';
  * Internal dependencies
  */
 import config from 'calypso/config';
-import userFactory from 'calypso/lib/user';
-import canCurrentUserUseCustomerHome from 'calypso/state/sites/selectors/can-current-user-use-customer-home';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { requestSite } from 'calypso/state/sites/actions';
 import getPrimarySiteId from 'calypso/state/selectors/get-primary-site-id';
-import { getSiteSlug, isJetpackSite } from 'calypso/state/sites/selectors';
-import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
+import { canCurrentUserUseCustomerHome, getSite, getSiteSlug } from 'calypso/state/sites/selectors';
 
 export default function () {
-	const user = userFactory();
-	const isLoggedOut = ! user.get();
-	if ( isLoggedOut ) {
-		setupLoggedOut();
-	} else {
-		setupLoggedIn();
-	}
+	page( '/', ( context ) => {
+		const isLoggedIn = isUserLoggedIn( context.store.getState() );
+		if ( isLoggedIn ) {
+			handleLoggedIn( context );
+		} else {
+			handleLoggedOut( context );
+		}
+	} );
 }
 
-function setupLoggedOut() {
+function handleLoggedOut() {
 	if ( config.isEnabled( 'desktop' ) ) {
-		page( '/', () => {
-			if ( config.isEnabled( 'oauth' ) ) {
-				page.redirect( '/oauth-login' );
-			} else {
-				page.redirect( '/log-in' );
-			}
-		} );
+		if ( config.isEnabled( 'oauth' ) ) {
```

Comment:
Should we be using `config( 'login_url` )` here in? It seems the different contexts have this variable defined in the  `config` directory.

Reply:
I replaced the `/oauth-login` string with `config( 'login_url' )` here. However, the `login_url` variable has somewhat different meanings in different contexts, making its usage a bit confusing. We should clean that up someday 🙂